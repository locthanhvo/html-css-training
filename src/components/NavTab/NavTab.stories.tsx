import { BrowserRouter } from 'react-router-dom';
import { Meta, StoryObj } from '@storybook/react';
import { Box } from '@chakra-ui/react';

// Components
import NavTab from '.';

// Icons
import {
  BarChartIcon,
  CalendarIcon,
  ChecklistIcon,
  HamburgerMenuIcon,
  StatisticsIcon,
  BurgerMenuLeftIcon,
} from '@/components/common/Icons';

const meta: Meta<typeof NavTab> = {
  title: 'Components/NavTab',
  component: NavTab,
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    ),
  ],
};

export default meta;

type Story = StoryObj<typeof NavTab>;

export const Primary: Story = {
  args: {
    tabList: [
      {
        label: 'List Tasks',
        component: <Box>Content</Box>,
        icon: ChecklistIcon,
      },
      {
        label: 'Boards',
        component: <Box>Content 1</Box>,
        icon: BarChartIcon,
      },
      {
        label: 'Calendar',
        component: <Box>Content 1</Box>,
        icon: CalendarIcon,
      },
      {
        label: 'Gantt',
        component: <Box>Content 1</Box>,
        icon: BurgerMenuLeftIcon,
      },
      {
        label: 'Timeline',
        component: <Box>Content 1</Box>,
        icon: HamburgerMenuIcon,
      },
      {
        label: 'Activity',
        component: <Box>Content 1</Box>,
        icon: StatisticsIcon,
      },
    ],
  },
};
