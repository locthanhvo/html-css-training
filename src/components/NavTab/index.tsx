import { memo, useState } from 'react';
import { Tabs, TabList, TabPanels, Tab, TabPanel } from '@chakra-ui/react';

interface TabProps {
  tabList: {
    label: string;
    component: JSX.Element;
  }[];
}

const TabLayout = ({ tabList }: TabProps) => {
  const [tabIndex, setTabIndex] = useState(0);

  const handleTabChange = (index: number) => {
    setTabIndex(index);
  };

  return (
    <Tabs>
      <TabList maxW="578px">
        {tabList.map((tab, index) => (
          <Tab
            px={4}
            py={3}
            bgColor={tabIndex === index ? 'white' : ''}
            color={tabIndex === index ? 'blue.200' : 'gray.500'}
            fontSize="base"
            fontWeight="regular"
            key={index}
            onClick={() => handleTabChange(index)}
          >
            {tab.label}
          </Tab>
        ))}
      </TabList>
      <TabPanels bgColor="white" borderColor="gray.200">
        {tabList.map((tab, index) => (
          <TabPanel key={index}>{tabIndex === index && tab.component}</TabPanel>
        ))}
      </TabPanels>
    </Tabs>
  );
};

export default memo(TabLayout);
