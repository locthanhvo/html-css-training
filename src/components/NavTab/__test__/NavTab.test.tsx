import { Box } from '@chakra-ui/react';
import { render, RenderResult } from '@testing-library/react';

// Components
import NavTab from '..';

// Icons
import {
  BarChartIcon,
  CalendarIcon,
  ChecklistIcon,
  HamburgerMenuIcon,
  StatisticsIcon,
  BurgerMenuLeftIcon,
} from '@/components/common/Icons';

describe('NavTab Component', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(
      <NavTab
        tabList={[
          {
            label: 'List Tasks',
            component: <Box>Content</Box>,
            icon: ChecklistIcon,
          },
          {
            label: 'Boards',
            component: <Box>Boards Content</Box>,
            icon: BarChartIcon,
          },
          {
            label: 'Calendar',
            component: <Box>Content 1</Box>,
            icon: CalendarIcon,
          },
          {
            label: 'Gantt',
            component: <Box>Content 1</Box>,
            icon: BurgerMenuLeftIcon,
          },
          {
            label: 'Timeline',
            component: <Box>Content 1</Box>,
            icon: HamburgerMenuIcon,
          },
          {
            label: 'Activity',
            component: <Box>Content 1</Box>,
            icon: StatisticsIcon,
          },
        ]}
      />,
    );
  });

  it('Should match snapshot', () => {
    const { container } = renderResult;
    expect(container).toMatchSnapshot();
  });

  it('should render tab correctly', () => {
    const { getByText } = renderResult;

    getByText('Boards').click();

    expect(getByText('Boards Content')).toBeInTheDocument();
  });
});
