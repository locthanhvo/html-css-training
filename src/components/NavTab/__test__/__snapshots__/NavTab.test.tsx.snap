// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavTab Component Should match snapshot 1`] = `
<div>
  <div
    class="chakra-tabs css-79elbk"
  >
    <div
      class="css-8anfc0"
    >
      <div
        aria-orientation="horizontal"
        class="chakra-tabs__tablist css-vjq03p"
        role="tablist"
      >
        <button
          aria-controls="tabs-:r0:--tabpanel-0"
          aria-disabled="false"
          aria-selected="false"
          class="chakra-tabs__tab css-vdln8n"
          data-index="0"
          id="tabs-:r0:--tab-0"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <svg
            fill="none"
            height="31"
            viewBox="0 0 30 31"
            width="30"
          >
            <path
              clip-rule="evenodd"
              d="M16 12.5H6C5.45 12.5 5 12.95 5 13.5C5 14.05 5.45 14.5 6 14.5H16C16.55 14.5 17 14.05 17 13.5C17 12.95 16.55 12.5 16 12.5ZM16 8.5H6C5.45 8.5 5 8.95 5 9.5C5 10.05 5.45 10.5 6 10.5H16C16.55 10.5 17 10.05 17 9.5C17 8.95 16.55 8.5 16 8.5ZM6 18.5H12C12.55 18.5 13 18.05 13 17.5C13 16.95 12.55 16.5 12 16.5H6C5.45 16.5 5 16.95 5 17.5C5 18.05 5.45 18.5 6 18.5ZM25.21 14.71L25.3 14.8C25.69 15.19 25.69 15.82 25.3 16.21L19.72 21.8C19.33 22.19 18.7 22.19 18.31 21.8L15.22 18.71C15.0327 18.5232 14.9275 18.2695 14.9275 18.005C14.9275 17.7405 15.0327 17.4868 15.22 17.3L15.31 17.21C15.7 16.82 16.33 16.82 16.72 17.21L19.02 19.51L23.8 14.72C24.18 14.32 24.82 14.32 25.21 14.71Z"
              fill="#C3CAD9"
              fill-rule="evenodd"
            />
          </svg>
          List Tasks
        </button>
        <button
          aria-controls="tabs-:r0:--tabpanel-1"
          aria-disabled="false"
          aria-selected="true"
          class="chakra-tabs__tab css-1gpqhhf"
          data-index="1"
          id="tabs-:r0:--tab-1"
          role="tab"
          tabindex="0"
          type="button"
        >
          <svg
            fill="none"
            height="31"
            viewBox="0 0 30 31"
            width="30"
          >
            <path
              clip-rule="evenodd"
              d="M7 16.5H13C13.55 16.5 14 16.05 14 15.5V7.5C14 6.95 13.55 6.5 13 6.5H7C6.45 6.5 6 6.95 6 7.5V15.5C6 16.05 6.45 16.5 7 16.5ZM7 24.5H13C13.55 24.5 14 24.05 14 23.5V19.5C14 18.95 13.55 18.5 13 18.5H7C6.45 18.5 6 18.95 6 19.5V23.5C6 24.05 6.45 24.5 7 24.5ZM17 24.5H23C23.55 24.5 24 24.05 24 23.5V15.5C24 14.95 23.55 14.5 23 14.5H17C16.45 14.5 16 14.95 16 15.5V23.5C16 24.05 16.45 24.5 17 24.5ZM16 7.5V11.5C16 12.05 16.45 12.5 17 12.5H23C23.55 12.5 24 12.05 24 11.5V7.5C24 6.95 23.55 6.5 23 6.5H17C16.45 6.5 16 6.95 16 7.5Z"
              fill="royalBlue"
              fill-rule="evenodd"
            />
          </svg>
          Boards
        </button>
        <button
          aria-controls="tabs-:r0:--tabpanel-2"
          aria-disabled="false"
          aria-selected="false"
          class="chakra-tabs__tab css-vdln8n"
          data-index="2"
          id="tabs-:r0:--tab-2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <svg
            fill="none"
            height="31"
            viewBox="0 0 30 31"
            width="30"
          >
            <path
              clip-rule="evenodd"
              d="M19 16.5H16C15.45 16.5 15 16.95 15 17.5V20.5C15 21.05 15.45 21.5 16 21.5H19C19.55 21.5 20 21.05 20 20.5V17.5C20 16.95 19.55 16.5 19 16.5ZM19 6.5V7.5H11V6.5C11 5.95 10.55 5.5 10 5.5C9.45 5.5 9 5.95 9 6.5V7.5H8C6.89 7.5 6.01 8.4 6.01 9.5L6 23.5C6 24.6 6.89 25.5 8 25.5H22C23.1 25.5 24 24.6 24 23.5V9.5C24 8.4 23.1 7.5 22 7.5H21V6.5C21 5.95 20.55 5.5 20 5.5C19.45 5.5 19 5.95 19 6.5ZM21 23.5H9C8.45 23.5 8 23.05 8 22.5V12.5H22V22.5C22 23.05 21.55 23.5 21 23.5Z"
              fill="#C3CAD9"
              fill-rule="evenodd"
            />
          </svg>
          Calendar
        </button>
        <button
          aria-controls="tabs-:r0:--tabpanel-3"
          aria-disabled="false"
          aria-selected="false"
          class="chakra-tabs__tab css-vdln8n"
          data-index="3"
          id="tabs-:r0:--tab-3"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <svg
            fill="none"
            height="31"
            viewBox="0 0 30 31"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M7 21.5H11C11.55 21.5 12 21.05 12 20.5C12 19.95 11.55 19.5 11 19.5H7C6.45 19.5 6 19.95 6 20.5C6 21.05 6.45 21.5 7 21.5ZM6 10.5C6 11.05 6.45 11.5 7 11.5H23C23.55 11.5 24 11.05 24 10.5C24 9.95 23.55 9.5 23 9.5H7C6.45 9.5 6 9.95 6 10.5ZM7 16.5H17C17.55 16.5 18 16.05 18 15.5C18 14.95 17.55 14.5 17 14.5H7C6.45 14.5 6 14.95 6 15.5C6 16.05 6.45 16.5 7 16.5Z"
              fill="#C3CAD9"
              fill-rule="evenodd"
            />
          </svg>
          Gantt
        </button>
        <button
          aria-controls="tabs-:r0:--tabpanel-4"
          aria-disabled="false"
          aria-selected="false"
          class="chakra-tabs__tab css-vdln8n"
          data-index="4"
          id="tabs-:r0:--tab-4"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <svg
            fill="none"
            height="31"
            viewBox="0 0 30 31"
            width="30"
          >
            <path
              clip-rule="evenodd"
              d="M7 12.5C7.55 12.5 8 12.05 8 11.5C8 10.95 7.55 10.5 7 10.5C6.45 10.5 6 10.95 6 11.5C6 12.05 6.45 12.5 7 12.5ZM8 15.5C8 16.05 7.55 16.5 7 16.5C6.45 16.5 6 16.05 6 15.5C6 14.95 6.45 14.5 7 14.5C7.55 14.5 8 14.95 8 15.5ZM8 19.5C8 20.05 7.55 20.5 7 20.5C6.45 20.5 6 20.05 6 19.5C6 18.95 6.45 18.5 7 18.5C7.55 18.5 8 18.95 8 19.5ZM24 15.5C24 16.05 23.55 16.5 23 16.5H11C10.45 16.5 10 16.05 10 15.5C10 14.95 10.45 14.5 11 14.5H23C23.55 14.5 24 14.95 24 15.5ZM23 20.5C23.55 20.5 24 20.05 24 19.5C24 18.95 23.55 18.5 23 18.5H11C10.45 18.5 10 18.95 10 19.5C10 20.05 10.45 20.5 11 20.5H23ZM11 12.5C10.45 12.5 10 12.05 10 11.5C10 10.95 10.45 10.5 11 10.5H23C23.55 10.5 24 10.95 24 11.5C24 12.05 23.55 12.5 23 12.5H11Z"
              fill="#C3CAD9"
              fill-rule="evenodd"
            />
          </svg>
          Timeline
        </button>
        <button
          aria-controls="tabs-:r0:--tabpanel-5"
          aria-disabled="false"
          aria-selected="false"
          class="chakra-tabs__tab css-vdln8n"
          data-index="5"
          id="tabs-:r0:--tab-5"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <svg
            fill="none"
            height="31"
            viewBox="0 0 30 31"
            width="30"
          >
            <path
              clip-rule="evenodd"
              d="M19.4325 10.35L20.8725 11.79L15.9925 16.67L12.7025 13.38C12.5157 13.1927 12.262 13.0875 11.9975 13.0875C11.733 13.0875 11.4793 13.1927 11.2925 13.38L5.2925 19.39C4.9025 19.78 4.9025 20.41 5.2925 20.8C5.6825 21.19 6.3125 21.19 6.7025 20.8L11.9925 15.5L15.2825 18.79C15.6725 19.18 16.3025 19.18 16.6925 18.79L22.2825 13.21L23.7225 14.65C24.0325 14.96 24.5725 14.74 24.5725 14.3V10C24.5825 9.72 24.3625 9.5 24.0825 9.5H19.7925C19.3425 9.5 19.1225 10.04 19.4325 10.35Z"
              fill="#C3CAD9"
              fill-rule="evenodd"
            />
          </svg>
          Activity
        </button>
      </div>
      <div
        class="css-k008qs"
      >
        <div
          class="chakra-form-control css-17qx7uq"
          role="group"
        >
          <div
            class="chakra-input__group css-bx0blc"
            data-group="true"
          >
            <div
              class="chakra-input__left-element css-1yuzhmq"
            >
              <svg
                fill="none"
                height="31"
                viewBox="0 0 30 31"
                width="30"
              >
                <path
                  clip-rule="evenodd"
                  d="M18.5006 17.5006H17.7106L17.4306 17.2306C18.6306 15.8306 19.2506 13.9206 18.9106 11.8906C18.4406 9.11063 16.1206 6.89063 13.3206 6.55063C9.09063 6.03063 5.53063 9.59063 6.05063 13.8206C6.39063 16.6206 8.61063 18.9406 11.3906 19.4106C13.4206 19.7506 15.3306 19.1306 16.7306 17.9306L17.0006 18.2106V19.0006L21.2506 23.2506C21.6606 23.6606 22.3306 23.6606 22.7406 23.2506C23.1506 22.8406 23.1506 22.1706 22.7406 21.7606L18.5006 17.5006ZM12.5006 17.5006C10.0106 17.5006 8.00063 15.4906 8.00063 13.0006C8.00063 10.5106 10.0106 8.50063 12.5006 8.50063C14.9906 8.50063 17.0006 10.5106 17.0006 13.0006C17.0006 15.4906 14.9906 17.5006 12.5006 17.5006Z"
                  fill="#C3CAD9"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <input
              class="chakra-input css-1iib86v"
              id="field-:r1:"
              placeholder="Search Tasks"
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="chakra-tabs__tab-panels css-w464rq"
    >
      <div
        aria-labelledby="tabs-:r0:--tab-0"
        class="chakra-tabs__tab-panel css-sjt5nk"
        hidden=""
        id="tabs-:r0:--tabpanel-0"
        role="tabpanel"
        tabindex="0"
      />
      <div
        aria-labelledby="tabs-:r0:--tab-1"
        class="chakra-tabs__tab-panel css-sjt5nk"
        id="tabs-:r0:--tabpanel-1"
        role="tabpanel"
        tabindex="0"
      >
        <div
          class="css-0"
        >
          Boards Content
        </div>
      </div>
      <div
        aria-labelledby="tabs-:r0:--tab-2"
        class="chakra-tabs__tab-panel css-sjt5nk"
        hidden=""
        id="tabs-:r0:--tabpanel-2"
        role="tabpanel"
        tabindex="0"
      />
      <div
        aria-labelledby="tabs-:r0:--tab-3"
        class="chakra-tabs__tab-panel css-sjt5nk"
        hidden=""
        id="tabs-:r0:--tabpanel-3"
        role="tabpanel"
        tabindex="0"
      />
      <div
        aria-labelledby="tabs-:r0:--tab-4"
        class="chakra-tabs__tab-panel css-sjt5nk"
        hidden=""
        id="tabs-:r0:--tabpanel-4"
        role="tabpanel"
        tabindex="0"
      />
      <div
        aria-labelledby="tabs-:r0:--tab-5"
        class="chakra-tabs__tab-panel css-sjt5nk"
        hidden=""
        id="tabs-:r0:--tabpanel-5"
        role="tabpanel"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;
